cmd/mesos_metrics_test.go

```diff
@@ -0,0 +1,98 @@
+package cmd
+
+import "testing"
+import "context"
+import "io"
+import "net"
+import "net/http"
+import "net/http/httptest"
+
+// TestMesosMetricsCheckUrl verifies we get the right url
+func TestMesosMetricsCheckUrl(t *testing.T) {
+	test := &MesosMetricsCheck{
+		Name: "TEST",
+	}
+
+	for _, testCase := range []struct {
+		role     string
+		forceTLS bool
+		expected string
+	}{
+		{
+			role:     "master",
+			forceTLS: false,
+			expected: "http://127.0.0.1:5050/metrics/snapshot",
+		},
+		{
+			role:     "agent_public",
+			forceTLS: false,
+			expected: "http://127.0.0.1:5051/metrics/snapshot",
+		},
+		{
+			role:     "agent",
+			forceTLS: true,
+			expected: "https://127.0.0.1:5051/metrics/snapshot",
+		},
+	} {
+		mockCLICfg := &CLIConfigFlags{
+			NodeIPStr: "127.0.0.1",
+			Role:      testCase.role,
+			ForceTLS:  testCase.forceTLS,
+		}
+
+		url, err := test.getURL(nil, mockCLICfg)
+		if err != nil {
+			t.Fatalf("Error running getURL: %s", err)
+		}
+
+		if url.String() != testCase.expected {
+			t.Fatalf("Expect %s. Got %s", testCase.expected, url)
+		}
+	}
+}
+
+// TestMesosMetricsCheckRun checks run
+func TestMesosMetricsCheckRun(t *testing.T) {
+	test := &MesosMetricsCheck{
+		Name: "TEST",
+	}
+
+	masterHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// A very simple health check.
+		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Content-Type", "application/json")
+
+		// In the future we could report back on the status of our DB, or our cache
+		// (e.g. Redis) by performing a simple PING, and include them in the response.
+		io.WriteString(w, `{"slave\/tasks_finished":0.0,"slave\/cpus_total":4.0,"slave\/executors_preempted":0.0,"registrar\/log\/recovered": 1.0}`)
+	})
+
+	masterServer := httptest.NewUnstartedServer(masterHandler)
+	l, _ := net.Listen("tcp", "127.0.0.1:5050")
+	masterServer.Listener = l
+	masterServer.Start()
+	defer masterServer.Close()
+
+	for _, testCase := range []struct {
+		role     string
+		forceTLS bool
+		expected string
+	}{
+		{
+			role:     "master",
+			forceTLS: false,
+			expected: "http://127.0.0.1:5050/metrics/snapshot",
+		},
+	} {
+		mockCLICfg := &CLIConfigFlags{
+			NodeIPStr: "127.0.0.1",
+			Role:      testCase.role,
+			ForceTLS:  testCase.forceTLS,
+		}
+
+		_, _, err := test.Run(context.TODO(), mockCLICfg)
```

Comment:
I think we need to be checking the return values of the runner other than just the error.